strapi/strapi
# [Custom fields] Fix number types

### What does it do
- Creates a dedicated action for custom fields
- Fixes an issue with number types not being able to be used since `default: null` was always set

### How to test it
- Register any number type, like `integer`

```js
// admin
app.customFields.register({
  type: 'integer',
  pluginId: 'myplugin',
  name: `custominteger`,
  intlLabel: {
    id: 'customfieldtest',
    defaultMessage: `custominteger`,
  },
  intlDescription: {
    id: 'customfieldtest',
    defaultMessage: `custominteger`,
  },
  components: {
    Input: async () =>
      import(/* webpackChunkName: "test-custom-field" */ './components/PluginIcon'),
  },
  options: {
    defaults: {},
  },
});

// server
strapi.customFields.register({
  type: 'integer',
  name: `custominteger`,
  plugin: 'myplugin',
});

```
- Go to the CTB and select your custom field
- Give it a name, and click finish
- It should be added to the CTB attributes list
